plugin-git/che-plugin-git-ext-git/src/main/java/org/eclipse/che/ide/ext/git/client/AddToIndexManager.java

```diff
@@ -0,0 +1,318 @@
+/*******************************************************************************
+ * Copyright (c) 2012-2016 Codenvy, S.A.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *   Codenvy, S.A. - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.che.ide.ext.git.client;
+
+import com.google.web.bindery.event.shared.EventBus;
+
+import org.eclipse.che.api.git.gwt.client.GitServiceClient;
+import org.eclipse.che.api.git.shared.Status;
+import org.eclipse.che.api.workspace.shared.dto.ProjectConfigDto;
+import org.eclipse.che.ide.api.app.AppContext;
+import org.eclipse.che.ide.event.CreateFileNodeEvent;
+import org.eclipse.che.ide.event.CreateFileNodeEventHandler;
+import org.eclipse.che.ide.event.DeleteNodesEvent;
+import org.eclipse.che.ide.event.DeleteNodesEventHandler;
+import org.eclipse.che.ide.event.RenameNodeEvent;
+import org.eclipse.che.ide.event.RenameNodeEventHandler;
+import org.eclipse.che.ide.api.notification.Notification;
+import org.eclipse.che.ide.api.notification.NotificationManager;
+import org.eclipse.che.ide.api.notification.StatusNotification;
+import org.eclipse.che.ide.ext.java.client.event.RefactoringProvidedEvent;
+import org.eclipse.che.ide.ext.java.client.event.RefactoringProvidedEventHandler;
+import org.eclipse.che.ide.ext.java.shared.dto.refactoring.ChangeInfo;
+import org.eclipse.che.ide.project.node.FileReferenceNode;
+import org.eclipse.che.ide.project.node.FolderReferenceNode;
+import org.eclipse.che.ide.project.node.ModuleNode;
+import org.eclipse.che.ide.project.node.ResourceBasedNode;
+import org.eclipse.che.ide.rest.AsyncRequestCallback;
+import org.eclipse.che.ide.rest.DtoUnmarshallerFactory;
+import org.eclipse.che.ide.ui.dialogs.CancelCallback;
+import org.eclipse.che.ide.ui.dialogs.ConfirmCallback;
+import org.eclipse.che.ide.ui.dialogs.DialogFactory;
+import org.eclipse.che.ide.websocket.WebSocketException;
+import org.eclipse.che.ide.websocket.rest.RequestCallback;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+
+import static org.eclipse.che.ide.api.notification.StatusNotification.Status.FAIL;
+import static org.eclipse.che.ide.api.notification.StatusNotification.Status.SUCCESS;
+
+/**
+ * @author Igor Vinokur
+ */
+public class AddToIndexManager implements CreateFileNodeEventHandler, RefactoringProvidedEventHandler, DeleteNodesEventHandler,
```

Comment:
@azatsarynnyy may be it would be better to rename this class simply to `IndexManager` and create several usefull methods to control index state? I suppose that we will have methods like `removeFromIndex` also in future...